__test__/row.test.ts

```
@@ -99,6 +99,193 @@ describe("generateIdForRow", () => {
+                if (key === "MEMBER_MAP") {
+                    return MOCK_MEMBER_MAP;
+
+                } else if (key === "GROUPS_MAP") {
+                    return MOCK_GROUPS_MAP;
+                } 
+            }),
+        }));
+
+        const { getAttendeesFromRow } = require("../src/main/row");
+
+        const result = getAttendeesFromRow(row);
+
+        expect(result).toEqual(['Jack Zhang', 'Angel Zhang']);
+    });
+
+    it('should return all specified domain members when no ministry names are present', () => {
+
+        interface Row {
+            [key: string]: any;
+        }
+
+        const row: Row = { domain: "INT'L", who: "" };
+
+        jest.mock('../src/main/propertiesService', () => ({
+            loadMapFromScriptProperties: jest.fn((key: string) => {
+                if (key === "MEMBER_MAP") {
+                    return MOCK_MEMBER_MAP;
+
+                } else if (key === "GROUPS_MAP") {
+                    return MOCK_GROUPS_MAP;
+                } 
+            }),
+        }));
+
+        const { getAttendeesFromRow } = require("../src/main/row");
+
+        const result = getAttendeesFromRow(row);
+
+        expect(result).toEqual(['Brian Lin', 'James Lee', 'Jack Zhang', 'Angel Zhang']);
+    });
+
+    it('should throw an error when both ministry and domain names are missing', () => {
+
+        interface Row {
+            [key: string]: any;
+        }
+
+        const row: Row = { domain: '', who: '' };
+
+        jest.mock('../src/main/propertiesService', () => ({
+            loadMapFromScriptProperties: jest.fn((key: string) => {
+                if (key === "MEMBER_MAP") {
+                    return MOCK_MEMBER_MAP;
+
+                } else if (key === "GROUPS_MAP") {
+                    return MOCK_GROUPS_MAP;
+                } 
+            }),
+        }));
+
+        const { getAttendeesFromRow } = require("../src/main/row");
+
+        expect(() => getAttendeesFromRow(row)).toThrow(
```

for now let's go with silent errors; i saw even today there's "no mbs" in a what column and nothing else filled out

---

Sounds good, I replaced the error with a console.log